src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,597 @@
+    }
+
+    (StrikeData memory lowerStrike, StrikeData memory higherStrike) = _validateAndSortStrikes(makerTrades);
+    _verifyHigherStrikeAmount(higherStrike.tradeAmount, extraData.length != 0);
+
+    _verifyRfqExecute(lowerStrike, higherStrike, maxFee);
+  }
+
+  /// @dev the following function validates whether the maker is selling or buying a spread
+  /// The following table determines whether or not the high strike should be sold or bought
+  ///
+  /// isCallSpread | isLongSpread | isTaker | makerShouldBeSellingSpread
+  ///      0       |       0      |    0    |   0 (higherStrike should be < 0)
+  ///      0       |       0      |    1    |   1 (higherStrike should be > 0)
+  ///      0       |       1      |    0    |   1
+  ///      0       |       1      |    1    |   0
+  ///      1       |       0      |    0    |   1
+  ///      1       |       0      |    1    |   0
+  ///      1       |       1      |    0    |   0
+  ///      1       |       1      |    1    |   1
+  function _verifyHigherStrikeAmount(int strikeAmount, bool isTaker) internal pure {
+    PPTSAStorage memory $ = _getPPTSAStorage();
+    bool makerShouldBeSellingSpread = ($.isCallSpread == $.isLongSpread) ? isTaker : !isTaker;
+    if (makerShouldBeSellingSpread && strikeAmount <= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    } else if (!makerShouldBeSellingSpread && strikeAmount >= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    }
+  }
+
+  function _validateAndSortStrikes(IRfqModule.TradeData[] memory makerTrades)
+    internal
+    view
+    returns (StrikeData memory lowerStrike, StrikeData memory higherStrike)
+  {
+    if (makerTrades.length != 2) {
+      revert PPT_InvalidParams();
+    }
+    StrikeData memory strike1 = _validateAndCreateIndividualStrike(makerTrades[0]);
+    StrikeData memory strike2 = _validateAndCreateIndividualStrike(makerTrades[1]);
+
+    if (
+      makerTrades[0].asset != makerTrades[1].asset || makerTrades[0].asset != address(_getPPTSAStorage().optionAsset)
+        || strike1.expiry != strike2.expiry
+    ) {
+      revert PPT_InvalidParams();
+    }
+
+    if (strike1.strike > strike2.strike) {
+      return (strike2, strike1);
+    }
+    return (strike1, strike2);
+  }
+
+  function _validateAndCreateIndividualStrike(IRfqModule.TradeData memory trade)
+    internal
+    view
+    returns (StrikeData memory)
+  {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    (uint optionExpiry, uint optionStrike, bool isCall) = OptionEncoding.fromSubId(trade.subId.toUint96());
+
+    if (
+      optionExpiry < block.timestamp + $.ppParams.optionMinTimeToExpiry
+        || optionExpiry > block.timestamp + $.ppParams.optionMaxTimeToExpiry
+    ) {
+      revert PPT_OptionExpiryOutOfBounds();
+    }
+
+    (uint callPrice, uint putPrice,) = _getOptionPrice(optionExpiry, optionStrike);
+    uint markPrice = isCall ? callPrice : putPrice;
+    if ($.isCallSpread != isCall) {
```

Comment:
could move this check above getting the call price (earlier reverts are generally better for debugging and gas saving)

Could even merge the check with the expiry check to save on contract size (rename error to `PPT_InvalidOptionDetails` or similar)

Reply:
Done